extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -149,16 +150,58 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
       return;
     }
 
-    this.shoppingTagEl_ = this.renderShoppingTagTemplate_();
     this.onRtlStateUpdate_(this.storeService_.get(StateProperty.RTL_STATE));
-    this.mutateElement(() => {
-      createShadowRootWithStyle(
-        this.element,
-        this.shoppingTagEl_,
-        shoppingTagCSS
-      );
-      this.hasAppendedInnerShoppingTagEl_ = true;
-    });
+
+    this.measureMutateElement(
+      () => {
+        this.shoppingTagEl_ = this.renderShoppingTagTemplate_();
+      },
+      () => {
+        createShadowRootWithStyle(
+          this.element,
+          this.shoppingTagEl_,
+          shoppingTagCSS
+        );
+        this.hasAppendedInnerShoppingTagEl_ = true;
+
+        const rightEdgeOfScreentoTagDistance =
+          document.getElementsByTagName('amp-story-shopping-attachment')[0]
+            ./*OK*/ clientWidth *
+            0.5 -
+          this.element./*OK*/ offsetLeft -
+          this.element./*OK*/ offsetWidth * 0.5;
```

Comment:
These will trigger relayouts and is precisely what has to be in a `measureElement`. It forces the browser to recalculate all elements positions after it's already done it so it can give you accurate measures.